src/twisted/test/test_ftp.py

```diff
@@ -1298,6 +1568,74 @@ def checkDownload(download):
 
 
 
+class FTPServerEprtDataConnectionTests(FTPServerPasvDataConnectionTests):
+    def setUp(self):
+        self.dataPorts = []
+        return FTPServerPasvDataConnectionTests.setUp(self)
+
+
+    def _makeDataConnection(self, ignored=None):
+        # Establish an active data connection (i.e. server connecting to
+        # client).
+        deferred = defer.Deferred()
+
+        class DataFactory(protocol.ServerFactory):
+
+            protocol = _BufferingProtocol
+
+            def buildProtocol(self, addr):
+                p = protocol.ServerFactory.buildProtocol(self, addr)
+                reactor.callLater(0, deferred.callback, p)
+                return p
+
+        dataPort = reactor.listenTCP(0, DataFactory(), interface='127.0.0.1')
+        self.dataPorts.append(dataPort)
+        cmd = 'EPRT |1|127.0.0.1|%s|' % dataPort.getHost().port
+        self.client.queueStringCommand(cmd)
+        return deferred
+
+
+    def tearDown(self):
+        """
+        Tear down the connection.
+
+        @return: L{defer.DeferredList}
+        """
+        dl = [defer.maybeDeferred(port.stopListening)
+              for port in self.dataPorts]
+        d = defer.maybeDeferred(
+            FTPServerPasvDataConnectionTests.tearDown, self)
+        dl.append(d)
+        return defer.DeferredList(dl, fireOnOneErrback=True)
+
+
+    def test_EPRTCannotConnect(self):
+        """
+        Listen on a port, and immediately stop listening as a way to find a
```

Comment:
Can you describe this in terms of what's being verified rather than what the setup does?  ( https://jml.io/pages/test-docstrings.html is the canonical guideline. )